lib/linux.js

```diff
@@ -59,11 +80,25 @@ module.exports = async () => {
 		execFile(xwininfoBin, xpropDetailsArgs.concat([activeWindowId]))
 	]);
 
-	return parseLinux({
+	const data = parseLinux({
 		activeWindowId,
 		boundsStdout,
 		stdout
 	});
+	if (data.owner && typeof data.owner.processId === 'number') {
+		try {
+			const [memoryUsage, path] = await Promise.all([
+				getMemoryUsageByPid(data.owner.processId),
+				getPathByPid(data.owner.processId)
+			]);
+			data.memoryUsage = memoryUsage;
+			data.owner.path = path;
+		} catch (error) {
+			// Would mean process died already, so we just ignore it
```

Comment:
You should not silently ignore the error.

Reply:
Ignoring in this case means not setting related data. If we handle it in a different way(say, throw an error), we will need to make the handing consistent(say, check if process is still alive when we return all the data, which will still be inconsistent, because it may die while we _return results_, so we would need to prevent process from getting killed until we returned).

In other words, there is only 2 ways we can effectively treat such problems: 
 - treat data as possibly partially available(so signal problem by leaving null/undefined)
 - throw an error

If we want to go with the second way, we have to rewrite most of this library to report errors on every piece of info that might be missing(and literally every piece we return here might be missing).

Reply:
We just return `undefined` if not all the data is available.

Reply:
Well it's implicitly undefined right now. You want me to add explicit properties with undefined values?

Reply:
I think I didn't make myself clear enough. Sorry about that. I mean that if we can't return *all* the data, we shouldn't return anything at all. This is how the macOS version works. See https://github.com/sindresorhus/active-win/commit/f06f7e30b1765be4de87c6b41802126cf3fcf897. So if all the data is not available, the value returned is just `undefined`.

Reply:
^ done

Reply:
@sindresorhus It is not a problem to implement this behaviour, but... Under windows it' should be fairly easy(though I don't want to try and reproduce that, I've had enough time with win32 api to hate it) to run in circumstances when part of that info isn't accessible, mostly due to lacking right or some kind of app protection. Under Linux, the system might be configured in such a way that part of this information is not available, some applications might not supply this information, plus user might not have rights to access this information(additionally to us handling it as if there being a window means there is some application which created it, which isn't the way X server works, as this application might be run on completely different machine, even working with different OS). Given all this and unfeasibility of accounting for all the possible cases without growing this lib to enormous sizes and complexity, wouldn't it be better to just go with best effort solution?

Reply:
"Best effort" means putting the burden on the user. If all properties can possibly return `undefined`, it means the user has to carefully handle each case. That's a nightmare in practise as a user. You don't really know when, what, and why something would be `undefined`.

I think we at least should make it so that either you get `owner` or you don't. So not make the individual properties of `owner` optionals.

Reply:
yeah, but going with 
```
	{
		title: 'Unicorns - Google Search',
		id: 5762,
		bounds: {
			x: 0,
			y: 0,
			height: 900,
			width: 1440
		},
		owner: {
			name: 'Google Chrome',
			processId: 310,
			bundleId: 'com.google.Chrome',
			path: '/Applications/Google Chrome.app'
		},
		memoryUsage: 11015432
	}
```
`title` might be not set/be meaningless on linux(experienced it while writing scripts for window manager), can frequently be not set on windows(a lot of apps prefer to render their own header, what always exists is window class).
`bounds`- might not be present on windows due to not having rights to read it(I think I've encountered such case)
`owner.name` - if i'm correct in mapping it to OS level stuff, might be anything on linux, processes can change it at will in runtime, probably same under windows, plus below.
`owner.processId` - for linux, as described above, can be not set, or be a process on another machine, or actually (and it's perfectly valid) be be any identificator usable for the purposes of signalling the process to stop(which is what it exists there for), And with incorrect(for our purposes, but correct for xorg purposes) pid, which still is a number, all the other data might be perceived by us as existing and valid, while actually being just data of some random process. For windows can map to process which handles creating windows, not to actual process which creates them(it was a long time ago, but as far as I remember there was something like that in activex stuff)
`owner.path` - above applies, at least under linux we might have no rights to read it
`memoryUsage` - above applies, plus both under windows and linux we might have no rights to read it.

That's not even counting timing errors, as in timing between acquiring the window and resolving info.

All in all, it's fairly complex to even catch all the cases when we're in a situations we _suspect_ we might not have data/have wrong data.

So if you ask me, I'd provide a flag that information might be incomplete/wrong at best, while still returning what we managed to gather. After all, most users probably don't use the lib for system critical stuff, and only care about part of data it provides.

Though in the end, it's your lib, and if you feel like it should be done this way(though probably returning `null`, not `undefined`, or maybe even throwing an error as it's supposed to be relatively rare case), I will.